lib/sql.js

```diff
@@ -29,78 +49,139 @@ SqlConnector.prototype.getTypes = function() {
   return ['db', 'rdbms', 'sql'];
 };
 
-/*!
+/**
  * Get the default data type for ID
+ * @param prop Property definition
  * Returns {Function}
  */
-SqlConnector.prototype.getDefaultIdType = function() {
+SqlConnector.prototype.getDefaultIdType = function(prop) {
+  /*jshint unused:false */
   return Number;
 };
 
-SqlConnector.prototype.query = function () {
-  throw new Error('query method should be declared in connector');
-};
-
-SqlConnector.prototype.command = function (sql, params, callback) {
-  return this.query(sql, params, callback);
+/**
+ * Get the default database schema name
+ * @returns {string} The default schema name, such as 'public' or 'dbo'
+ */
+SqlConnector.prototype.getDefaultSchemaName = function() {
+  return '';
 };
 
-SqlConnector.prototype.queryOne = function (sql, callback) {
-  return this.query(sql, function (err, data) {
-    if (err) {
-      return callback(err);
-    }
-    callback(err, data && data[0]);
-  });
+/**
+ * Get the database schema name for the given model. The schema name can be
+ * customized at model settings or connector configuration level as `schema` or
+ * `schemaName`. For example,
+ *
+ * ```json
+ * "Customer": {
+ *   "name": "Customer",
+ *   "mysql": {
+ *     "schema": "MYDB",
+ *     "table": "CUSTOMER"
+ *   }
+ * }
+ * ```
+ *
+ * @param {String} model The model name
+ * @returns {String} The database schema name
+ */
+SqlConnector.prototype.schema = function(model) {
+  // Check if there is a 'schema' property for postgresql
+  var settings = this.getModelDefinition(model).settings || {};
+  var dbMeta = settings && settings[this.name];
```

Comment:
I see that `settings[this.name]` is used in multiple places, please consider creating a helper function.

An example implementation to illustrate my point:

```js
SqlConnector.prototype._getConnectorSpecificSettings = function(model) {
  var settings = this.getModelDefinition(model).settings || {};
  return settings[this.name];
}
```